File: 3184/main.go
package main

import "fmt"

func main() {
	fmt.Println(countCompleteDayPairs([]int{12,12,30,24,24}))
	fmt.Println(countCompleteDayPairs([]int{72, 48, 24, 3}))

}

func countCompleteDayPairs(hours []int) int {
	n := len(hours)
	ans := 0
	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			if (hours[i] + hours[j]) % 24 == 0 {
				ans++
			}
		}
	}
	return ans
}

File: 3184/main.rs
struct Solution {}

impl Solution {
    pub fn count_complete_day_pairs(hours: Vec<i32>) -> i32 {
        let n = hours.len();
        let mut ans = 0;
        for i in 0..n {
            for j in i+1..n {
                if (hours[i] + hours[j]) % 24 == 0 {
                    ans+=1;
                }
            }
        }
        return ans;
    }
}

fn main() {
    println!("{}", Solution::count_complete_day_pairs(vec![12,12,30,24,24]));
}